tools/analyzer_plugin/test/unit/util/boilerplate_utils_test.dart

```diff
@@ -0,0 +1,282 @@
+      });
+
+      test('removes over_react part directive if it exists', () async {
+        final result = await parseAndGetResolvedUnit(sourceWithOverReactPart, path: 'foo.dart');
+        final sourceChange = await buildFileEdit(result, (builder) {
+          removeOverReactGeneratedPartDirective(builder, result.unit);
+        });
+        final editList = sourceChange.edits?.firstOrNull?.edits;
+
+        expect(editList, isNotNull);
+        expect(editList.length, 1, reason: 'there should be one edit in the file');
+
+        final part = getOverReactGeneratedPartDirective(result.unit);
+        expect(editList.first.offset, part.offset);
+        expect(editList.first.length, part.length);
+        expect(editList.first.replacement, '');
+      });
+    });
+
+    group('fixOverReactGeneratedPartDirective', () {
+      group('does nothing if', () {
+        test('there is no part directive in file', () async {
+          final sourceFileEdits = await getSourceFileEdits(
+            sourceWithNoPart,
+            (builder, result) => fixOverReactGeneratedPartDirective(builder, result.unit, result.uri),
+            path: 'foo.dart',
+          );
+          expect(sourceFileEdits, isEmpty);
+        });
+
+        test('part directive in file is valid', () async {
+          final sourceFileEdits = await getSourceFileEdits(
+            sourceWithOverReactPart,
+            (builder, result) => fixOverReactGeneratedPartDirective(builder, result.unit, result.uri),
+            path: 'foo.dart',
+          );
+          expect(sourceFileEdits, isEmpty);
+        });
+
+        test('part directive in file is not over_react', () async {
+          final sourceFileEdits = await getSourceFileEdits(
+            sourceWithNonOverReactPart,
+            (builder, result) => fixOverReactGeneratedPartDirective(builder, result.unit, result.uri),
+            path: 'foo.dart',
+          );
+          expect(sourceFileEdits, isEmpty);
+        });
+      });
+
+      test('replaces over_react part directive that does not match file uri', () async {
+        final result = await parseAndGetResolvedUnit(sourceWithOverReactPart, path: 'different_file_name.dart');
+        final sourceChange = await buildFileEdit(result, (builder) {
+          fixOverReactGeneratedPartDirective(builder, result.unit, result.uri);
+        });
+        final editList = sourceChange.edits?.firstOrNull?.edits;
+
+        expect(editList, isNotNull);
+        expect(editList.length, 1, reason: 'there should be one edit in the file');
+
+        final part = getOverReactGeneratedPartDirective(result.unit);
+        expect(editList.first.offset, part.offset);
+        expect(editList.first.length, part.length);
+        expect(editList.first.replacement, 'part \'different_file_name.over_react.g.dart\';');
+      });
+    });
+  });
+}
+
+Future<void> overReactGeneratedPartDirectiveIsValidTestHelper({
+  bool isPartOverReact,
+  String path,
+  bool expectedResult,
```

Comment:
#superdupernit Since there are three different `expects`, I think this name could be more descriptive to the actual `part` validation (`expectedPartValidity`, `partIsValid`, etc). Though, I know the scope of this utility is super limited so ignore me if I'm being too nit picky 😄